### PR TITLE
Support setting custom background color for syntax highlighting

### PR DESCRIPTION
**Pull request by DanielleMaywood**
Release Notes:

- Added support for `background_color` in `syntax` map in `theme.json`.

This adds support for setting a `background_color` for styles inside the `syntax` map for themes defined in `theme.json`. The field is optional so there should be no backwards compatibility issues.

It is worth noting that the current behaviour for selecting text is that the background colours will mix/blend (I'm not sure the correct term here). Changing this behaviour, or making it configurable, looks to be a far more complex issue and I'm not sure I know how to do it.

### CODE REVIEW

**Comment by DanielleMaywood** [comment]
@cla-bot check

File: crates/theme/src/schema.rs [hunk 1]
```diff
@@ -1309,11 +1313,17 @@ pub struct HighlightStyleContent {
 
     #[serde(deserialize_with = "treat_error_as_none")]
     pub font_weight: Option<FontWeightContent>,
+
+    #[serde(deserialize_with = "treat_error_as_none")]
+    pub background_color: Option<String>,
```
Comment (maxdeviant) [inline crates/theme/src/schema.rs:1318]: Very minor nit, but can we put the `background_color` field below the `color` field (both here in the `struct` decl, but also at the initialization sites)?

Comment (DanielleMaywood) [reply crates/theme/src/schema.rs:1318]: I can do, the reason I did the order this way is defined in `HighlightStyle`

https://github.com/zed-industries/zed/blob/fcb4abf18b20281502b48e8691beede2b37934d8/crates/gpui/src/style.rs#L297-L317

Comment (DanielleMaywood) [reply crates/theme/src/schema.rs:1318]: I've just realised the `font_weight` and `font_style` fields are flipped so I guess my reasoning doesn't really hold 😄

Comment (DanielleMaywood) [reply crates/theme/src/schema.rs]: @maxdeviant I've changed the order

**Review by maxdeviant — APPROVED** [review]
Thank you!

**Comment by meocoder31099** [comment]
I'm adding a background color to the Inlay hint, essentially similar to applying syntax highlighting. However, I'm encountering two issues with coloring, so I want to discuss it here to see if we can find a solution at the moment.

Please refer to the attached image to better understand the issues

### Should not be like this
- The background color of the two lines should be separate, there should be a gap between them
no_2
- The background color is applied to both the indentation margin, it should not be done like that
no

## Should be like this
yes_2
yes

**Comment by maxdeviant** [comment]
> Hi, I'm adding a background color to the Inlay hint (#12485), essentially similar to applying syntax highlighting. However, I'm encountering two issues with coloring, so I want to discuss it here to see if we can find a solution at the moment.
> 
> Please refer to the attached image to better understand the issues
> 
> ### Should not be like this
> * The background color of the two lines should be separate, there should be a gap between them
>   no_2
> * The background color is applied to both the indentation margin, it should not be done like that
>   no
> 
> ## Should be like this
> yes_2 yes

Could you open an issue for this?

**Comment by meocoder31099** [comment]
> Could you open an issue for this?

sure

**Pull request merged**